packages/squiggle-lang/src/rescript/Reducer/Reducer_ErrorValue.res

```diff
@@ -23,6 +30,9 @@ let errorToString = err =>
       }
       answer
     }
+  | REMacroNotFound(macro) => `Macro not found: ${macro}`
```

Comment:
what are macros in this context?

Reply:
Macros as in Lisp-macros. More explanation.  A macro its parameter type is Lisp-AST. In our cae expression tree. Instead of reduced values, macros process the expression tree's as parameters. And they insert the resulting expression tree. They have 2 uses.
- Create control structures like ifthenelse and tertiary operator. 
- Lazy boolean operators.
- Fuctions getting code blocks as parameters.
They are there when we need lazy building blocks.

Example:
$$$ithenlse( truth, code1, code2)

When you have a Haskell lime 100% lazy language, you dont need  macros. But it the language is eager then you need macros. Ruby, Elixir, Erlang, Lisp are all using this mechanism under different names. In all those languages you can create custom control structures.